### PR TITLE
refactor: refactor bfd static route tests

### PR DESCRIPTION
**Pull request by cyw233**
### Description of PR

This is a follow-up PR for #12995. This PR includes more code refactoring on `bfd/test_bfd_static_route.py`, such as parametrizing IPv4 and IPv6 for each test, moving static methods out of `BfdBase` class, removing unnecessary fixtures and etc.

Summary:
Fixes # (issue) Microsoft ADO 27749982

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?
To refactor the test cases in `bfd/test_bfd_static_route.py` to make it more readable and efficient

#### How did you verify/test it?
I have run all test cases in this file in a T2 chassis device and can confirm they all passed.

### CODE REVIEW

**Comment by wsycqyz** [comment]
This seems to be a huge change. Do you have chance to test on chassis testbed?

**Comment by cyw233** [comment]
> This seems to be a huge change. Do you have chance to test on chassis testbed?

Yes, I tested all test cases on a chassis testbed and can confirm they have passed

**Review by wsycqyz — APPROVED** [review]
LGTM

File: tests/bfd/test_bfd_static_route.py [hunk 1]
```diff
@@ -20,7 +22,7 @@ class TestBfdStaticRoute(BfdBase):
     TOTAL_ITERATIONS = 100
```
Comment (cyw233) [inline tests/bfd/test_bfd_static_route.py:22]: Currently we have 100 iterations for BFD flap test, which will take a long time (~4h) to complete. I will change the iteration number to be based on the completeness level in my next PR (Microsoft ADO 28301940)

**Review by abdosi — APPROVED** [review]

**Pull request merged**